device_discovery/check_devices.py

```diff
@@ -2,14 +2,33 @@
 
 import subprocess
 
+#runs the bash script and records all of the outputs
 devices = subprocess.run(["./check_arp.sh"], stdout=subprocess.PIPE)
 output = devices.stdout.decode()
 
-macs = []
+#set a flag to differentiate between eth0 and wlan0 when collecting addr
+isWlan = False;
+#create two lists that will hold our addr
+macsEth0 = []
+macsWlan0 = []
+#loop through the output and load up both lists with the addr
 for line in output.split('\n'):
     dev = line.split(' ')
-    if(len(dev) < 2):
-        continue
-    macs.append({"IPv4":dev[0], "MAC":dev[1]})
+   #use $ character as a delimiter between the two types of addr
+   if line == "$":
+        #when $ is found flip the boolean to True
+        isWlan = True
+    #If the $ character is not found load up the eth0 list
+    if isWlan == False:
+        if(len(dev) < 2):
```

Comment:
This is repeated code -- why not just have it once right after line 20?

Reply:
I think the idea behind doing it this way was so that it could continue to loop after the $ character and grab all of the wlan devices connected. Otherwise I'd have to have a $ before every wlan entry. Unless I'm misunderstanding what you're suggesting.

Reply:
You only need to see the $ once to set the flag
When seeing it the first time you can set the flag and then `continue`